modules/opening/src/main/OpeningSearch.scala

```diff
@@ -90,29 +88,31 @@ private object OpeningSearch:
       Entry(op, tokenize(op))
     }.toList
 
-  private def scoreOf(query: Query, entry: Entry): Option[Score] = {
+  private def scoreOf(query: Query, entry: Entry): Option[Score] =
+    val entryPgnLower            = entry.opening.pgn.value.toLowerCase()
+    val entryUciLower            = entry.opening.uci.value.toLowerCase()
+    val entryTokensLower         = entry.tokens.map(_.toLowerCase)
```

Comment:
tokens should already be lowercase https://github.com/lichess-org/lila/blob/master/modules/opening/src/main/OpeningSearch.scala#L58-L58

Can you add a test that shows they're not and that `entryTokensLower` is necessary?

Reply:
@ornicar Lowercase is guaranteed for query tokens, but the entry tokens from the DB can have uppercase chars. The new test that fails on the old implementation is `test("piece letter in notation")`.

Reply:
I tried `val entryTokensLower         = entry.tokens` and the tests are still passing.

Reply:
@ornicar My bad. I've added test query "Nc6" - if entry tokens aren't converted to lowercase, the test will now fail (only two openings are given).